python/seldon_deploy_sdk/auth/openid.py

```diff
@@ -14,14 +14,16 @@
 
 logger = logging.getLogger(__name__)
 
-IdTokenField = "id_token"
-
 
 class OIDCIntegration(FrameworkIntegration):
     oauth2_client_cls = OAuth2Session
 
 
 class OIDCAuthenticator(Authenticator):
+
+    _IdTokenField = "id_token"
+    _AuthCodeState = "sd-sdk-state"
```

Comment:
Is this state global on the provider? As in, could it cause issues if multiple users logged in at the same time from the SDK?

Reply:
This isn't specific to any OAuth/OIDC provider - the state is allowed to be any string according to the spec, which we're using to our advantage here

Reply:
To confirm, the OAuth provider should treat the `state` opaque and simply pass it on to the callback handler, so there should be no impact on multiple concurrent client sessions.

From https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest section 3.1.2.1:
> state
> RECOMMENDED. Opaque value used to maintain state between the request and the callback. Typically, Cross-Site Request Forgery (CSRF, XSRF) mitigation is done by cryptographically binding the value of this parameter with a browser cookie.